spm-monitor-generic/src/main/java/com/sematext/spm/client/functions/TrimTimeUnit.java

```diff
@@ -0,0 +1,127 @@
+package com.sematext.spm.client.functions;
+
+import com.sematext.spm.client.observation.CalculationFunction;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * - LongTrimTimeUnit - Trims the time unit strings from the end of value of the metricName
+ * and returns the value converted to specified time unit.
+ * E.g. LongTrimUnit(discoveryTime, ms) - trims the time unit from the end of value of metric `discoveryTime` and
+ * convert & return the result in milliseconds.
+ */
+public class TrimTimeUnit implements CalculationFunction {
```

Comment:
Should be abstract or it makes sense to allow its creation? Comment should maybe be moved to LongTrimTimeUnit?

Reply:
Could be made abstract. Missed during refactoring. will update